consensus/consensus_service.go

```diff
@@ -160,9 +160,11 @@ func (consensus *Consensus) ResetState() {
 
 // ToggleConsensusCheck flip the flag of whether ignore viewID check during consensus process
 func (consensus *Consensus) ToggleConsensusCheck() {
-	consensus.infoMutex.Lock()
-	defer consensus.infoMutex.Unlock()
-	consensus.ignoreViewIDCheck = !consensus.ignoreViewIDCheck
+	if consensus.ignoreViewIDCheck.IsSet() {
```

Comment:
There is a potential read write race condition here. Please use atomic.Bool.Toggle. abool seems not supporting toggle implementation.

Reply:
Toggle implementation (from go-uber):

```
func (b *Bool) Toggle() bool {
	for {
		old := b.Load()
		if b.CAS(old, !old) {
			return old
		}
	}
}
```

Reply:
there is a fork with toggle support. forked it to our repo.